src/MapboxStyleParser.ts

```diff
@@ -281,18 +304,31 @@ export class MapboxStyleParser implements StyleParser<Omit<MbStyle, 'sources'>>
    * @param spriteName Name of the sprite
    * @return the url that returns the single image
    */
-  getIconImage(spriteName: string): (string|undefined) {
+  async getIconImage(spriteName: string): Promise<Sprite | undefined> {
     if (!spriteName) {
       return;
     }
     if (!this.spriteBaseUrl) {
       return;
     }
-    // TODO update endpoint as soon as api specification was made
-    let url: string = '/sprites/?';
-    url += 'name=' + spriteName;
-    url += '&baseurl=' + encodeURIComponent(this.spriteBaseUrl);
-    return url;
+
+    const response = await fetch(this.spriteBaseUrl + '.json');
```

Comment:
I'm wondering if we can place the fetch request at a better position. Right now, we fetch the spritesheet for every distinct icon, which can lead to quite some requests. Maybe we can check before, if sprites are used and fetch once per `readStyle`/`writeStyle`. Or maybe we can do this when instantiating the parser. I think spritesheets might be quite static.